app/controllers/concerns/api/v2/parameter_deprecation.rb

```diff
@@ -0,0 +1,61 @@
+# This module adds a pre-action filter that will handle deprecated properties.
```

Comment:
In our case the parameter is not on the first level: it's `params[:override_values][:use_puppet_default]`.